tool/simulate-dialogs.ts

```diff
@@ -129,14 +199,53 @@ class SimulatorStream extends Stream.Transform {
         newTurn.agent = utterance;
         newTurn.agent_target = prediction.prettyprint();
 
-        return {
+        this.push({
             id: dlg.id,
             turns: dlg.concat([newTurn])
-        };
+        });
     }
 
     _transform(dlg : ParsedDialogue, encoding : BufferEncoding, callback : (err : Error|null, dlg ?: DialogueExample) => void) {
-        this._run(dlg).then((dlg) => callback(null, dlg), callback);
+        this._run(dlg).then(() => callback(null), callback);
+    }
+
+    _flush(callback : () => void) {
+        callback();
+    }
+}
+
+class DialogueToPartialDialoguesStream extends Stream.Transform {
+
+    constructor() {
+        super({ objectMode : true });
+    }
+
+    private copyDialogueTurns(turns : Array<DialogueTurn>) : Array<DialogueTurn> {
+        const copy = new Array<DialogueTurn>();
```

Comment:
`copy : DialogueTurn[] = []` is more readable and faster at runtime (due to JS interpreter shenanigans)